api/config/service/topology/v1/topology.proto

```diff
@@ -12,6 +12,8 @@ message Config {
 }
 
 message Cache {
-  // The time to live (ttl) for an item in cache, the default is 1 hour (3600 seconds).
-  google.protobuf.Duration ttl = 1 [ (validate.rules).duration.gt.seconds = 1 ];
+  // This disables the caching functionality completely, all other aspects of the topology will continue to function.
+  bool disable = 1;
+  // The time to live (ttl) for an item in cache, the default is 2 hours (7200 seconds).
+  google.protobuf.Duration ttl = 2 [ (validate.rules).duration.gt.seconds = 1 ];
```

Comment:
i wonder if we should use the presence of the cache message to enable and disable it...

currently this would require you to define the TTL even if you have the cache disabled for config validation to pass.

Reply:
Strange I didn't have TTL set in my testing locally and config validation passed. But  I like your idea better and will change it up.